src/transactions/test/ChangeTrustTests.cpp

```diff
@@ -298,11 +298,17 @@ TEST_CASE("change trust", "[tx][changetrust]")
             // trustlines will be deauthorized initially
             gateway.setOptions(setFlags(AUTH_REQUIRED_FLAG));
 
-            // root is missing trustline(s)
-            REQUIRE_THROWS_AS(root.changeTrust(poolShareAsset, 10),
-                              ex_CHANGE_TRUST_TRUST_LINE_MISSING);
+            bool rootIsIssuerA = isIssuer(root, assetA);
+            bool rootIsIssuerB = isIssuer(root, assetB);
 
-            if (assetA.type() != ASSET_TYPE_NATIVE)
+            if (!rootIsIssuerA || !rootIsIssuerB)
+            {
+                // root is missing trustline(s)
+                REQUIRE_THROWS_AS(root.changeTrust(poolShareAsset, 10),
+                                  ex_CHANGE_TRUST_TRUST_LINE_MISSING);
+            }
+
+            if (assetA.type() != ASSET_TYPE_NATIVE && !rootIsIssuerA)
```

Comment:
Nitpick: You're repeating these conditions a lot of times, might be worth just adding variables like `hasTrustA` and `hasTrustB` to avoid duplication.

Reply:
Done